src/Parser/VengeanceDemonHunter/CombatLogParser.js

```diff
@@ -0,0 +1,191 @@
+
+  generateResults() {
+    const results = super.generateResults();
+
+    const fightDuration = this.fightDuration;
+
+    const nonDpsTimePercentage = this.modules.alwaysBeCasting.totalDamagingTimeWasted / fightDuration;
+    const deadTimePercentage = this.modules.alwaysBeCasting.totalTimeWasted / fightDuration;
+
+    console.log(this.modules.alwaysBeCasting.totalTimeWasted+'---'+fightDuration);
+
+    if (nonDpsTimePercentage > 0.01) {
+      results.addIssue({
+        issue: `Your non DPS time can be improved. Try to cast damaging spells more regularly.`,
+        stat: `${Math.round(nonDpsTimePercentage * 100)}% non DPS time (<30% is recommended)`,
+        icon: 'petbattle_health-down',
+        importance: getIssueImportance(nonDpsTimePercentage, 0.4, 0.45, true),
+      });
+    }
+    if (deadTimePercentage > 0.2) {
+      results.addIssue({
+        issue: `Your dead GCD time can be improved. Try to Always Be Casting (ABC).`,
+        stat: `${Math.round(deadTimePercentage * 100)}% dead GCD time (<20% is recommended)`,
+        icon: 'spell_mage_altertime',
+        importance: getIssueImportance(deadTimePercentage, 0.35, 0.4, true),
+      });
+    }
+    const castEfficiency = getCastEfficiency(CPM_ABILITIES, this);
+    castEfficiency.forEach((cpm) => {
+      if (cpm.canBeImproved && !cpm.ability.noSuggestion) {
+        results.addIssue({
+          issue: <span>Try to cast <SpellLink id={cpm.ability.spell.id} /> more often ({cpm.casts}/{cpm.maxCasts} casts: {Math.round(cpm.castEfficiency * 100)}% cast efficiency). The recommended cast efficiency is {Math.round(cpm.recommendedCastEfficiency * 100)}%. {cpm.ability.extraSuggestion || ''}</span>,
+          icon: cpm.ability.spell.icon,
+          importance: cpm.ability.importance || getIssueImportance(cpm.castEfficiency, cpm.recommendedCastEfficiency - 0.05, cpm.recommendedCastEfficiency - 0.15),
+        });
+      }
+    });
+
+    results.statistics = [
+      <StatisticBox
+        icon={<Icon icon="class_demonhunter" alt="Damage taken" />}
+        value={`${formatNumber(this.totalDamageTaken / this.fightDuration * 1000)} DTPS`}
+        label='Damage taken'
+        tooltip={`Damage taken breakdown:
+            <ul>
+              ${Object.keys(this.damageBySchool).reduce((v, x) => {return v+=`<li>${getMagicDescription(x)} damage taken ${formatThousands(this.damageBySchool[x])} (${formatPercentage(this.damageBySchool[x]/this.totalDamageTaken)}%)</li>`; }, '')}
+            </ul>
+            Total damage taken ${formatThousands(this.totalDamageTaken)}`}
+      />,
+      <StatisticBox
+        icon={(
+          <img
+            src="/img/healing.png"
+            style={{ border: 0 }}
+            alt="Healing"
+          />
+        )}
+        value={`${formatNumber(this.totalHealing / this.fightDuration * 1000)} HPS`}
+        label='Healing done'
+        tooltip={`The total healing done was ${formatThousands(this.totalHealing)}.`}
+      />,
+      <StatisticBox
```

Comment:
With the new module setup I believe this belongs to the AlwaysBeCasting module in the `statistic()` function